rebac-admin-backend/v1/error.go

```diff
@@ -87,11 +90,23 @@ func NewDelegateErrorResponseMapper(m ErrorResponseMapper) ErrorResponseMapper {
 	return &delegateErrorResponseMapper{delegate: m}
 }
 
-// isBadRequestError determines whether the given error should be teated as a
-// "Bad Request" (400) error.
-func isBadRequestError(err error) bool {
+// mapHandlerInternalError checks if the given error is an internal
+// handler-level error (i.e., an auto-generated error type) and return the
+// equivalent errorWithStatus instance. If the given error is not an internal
+// handler error, this function will return nil.
+func mapHandlerInternalError(err error) *errorWithStatus {
 	// Note that these are all auto-generated error types, that should be
 	// regarded as bad request errors.
+	if !isHandlerInternalError(err) {
+		return nil
+	}
+	return &errorWithStatus{
+		status:  http.StatusBadRequest,
```

Comment:
question: internal error usually maps to 5xx, but this is for 400 bad requests, didn't the previous name of this function sound more appropriate?